plugins/kubernetes/app/models/kubernetes/role.rb

```diff
@@ -0,0 +1,23 @@
+module Kubernetes
+  class Role < ActiveRecord::Base
+    self.table_name = 'kubernetes_roles'
+    belongs_to :project, inverse_of: :roles
+
+    DEPLOY_STRATEGIES = ['rolling_update', 'kill_and_restart']
+
+    validates :project, presence: true
+    validates :name, presence: true
+    validates :deploy_strategy, presence: true, inclusion: DEPLOY_STRATEGIES
+    validates :replicas, presence: true, numericality: { greater_than: 0 }
```

Comment:
I can't think of a scenario where you'd want to define a role that had zero replicas.  If we come up with one, I'll change this validation.